src/libstd/sync/mpsc/mod.rs

```diff
@@ -1540,6 +1645,87 @@ mod tests {
     }
 
     #[test]
+    fn oneshot_single_thread_recv_timeout() {
+        let (tx, rx) = channel();
+        tx.send(()).unwrap();
+        assert_eq!(rx.recv_timeout(Duration::from_millis(100)), Ok(()));
+        assert_eq!(rx.recv_timeout(Duration::from_millis(100)), Err(RecvTimeoutError::Timeout));
+        tx.send(()).unwrap();
+        assert_eq!(rx.recv_timeout(Duration::from_millis(100)), Ok(()));
+    }
+
+    #[test]
+    fn stress_recv_timeout_two_threads() {
+        let (tx, rx) = channel();
+        let stress = stress_factor();
```

Comment:
This `stress_factor()` is actually almost always 1 nowadays, so perhaps a more explicit count could be used?